atomic_reactor/utils/resource_manager.py

```diff
@@ -0,0 +1,132 @@
+"""
+Copyright (c) 2022 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+import os
+import subprocess
+import logging
+import json
+from datetime import datetime
+from shlex import quote
+
+ROOT_PATH = 'resource_manager/'
+logger = logging.getLogger(__name__)
+
+
+class ResourceManager:
+    '''
+        Manages build resources
+    '''
+
+    def __init__(self):
+        pass
+
+    def check_avail(self, user, host):
+        try:
+            data = os.popen(f'ssh {user}@{host} -4C "cat {ROOT_PATH}info.json"').read()
+            if data == '':
+                logger.warning("%s hasn't been deployed yet", host)
+                return False
+            else:
+                # parse slot count, arch
+                meta = json.loads(data)
+                logger.debug("%s is up", meta['hostname'])
+
+                # check slots status
+                meta['slots'] = []
+                free_slots = 0
+                i = 0
+                while i < meta.get('max_slot_count', 1):
+                    slot = os.popen(f'ssh {user}@{host} -4C "cat {ROOT_PATH}slot_{i}.json"').read()
```

Comment:
I sugest to use helper method for this one for easier mocking

```
self._call_cmd(cmd)  # calls popen internally and returns result
```
with OOP changes I mentioned earlier, it can even construct `ssh {user}@{host} -4C` part and you can really pass only CMD itself